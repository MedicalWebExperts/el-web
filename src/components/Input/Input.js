import React, { Component } from 'react';
import { string, oneOf, func } from 'prop-types';

import theme from '../../styles';

class Input extends Component {
  static propTypes = {
    value: string,
    name: string.isRequired,
    type: oneOf(['text', 'email', 'tel', 'select']).isRequired,
    placeholder: string,
    onChange: func.isRequired,
  };

  static defaultProps = {
    value: 'Default Value',
    placeholder: 'Placeholder',
  };

  state = {
    value: this.props.value,
  };

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ value });
    this.props.onChange({ [name]: value });
  };

  render() {
    const { type, name, placeholder } = this.props;
    const { value } = this.state;
    return (
      <div>
        {
          <style jsx>
            {`
              input,
              select {
                padding: 10px;
                font-size: 14px;
                border-radius: 4px;
                border-width: 1px;
                border-style: solid;
                background-color: ${theme.colors.white};
                color: ${theme.colors.textTertiary};
                border-color: ${theme.colors.textTertiary};
                outline: none;
              }
              input:active,
              input:focus {
                outline: none;
              }
              input::placeholder {
                color: ${theme.colors.textTertiary};
                font-style: italic;
              }
            `}
          </style>
        }
        {type !== 'select' ? (
          <input
            type={type}
            name={name}
            value={value}
            onChange={this.handleChange}
            placeholder={placeholder}
          />
        ) : (
          <select type={type} name={name} value={value} onChange={this.handleChange}>
            <option value="volvo">Volvo</option>
            <option value="saab">Saab</option>
            <option value="mercedes">Mercedes</option>
            <option value="audi">Audi</option>
          </select>
        )}
      </div>
    );
  }
}

export default Input;
