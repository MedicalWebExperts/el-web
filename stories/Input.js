import React from 'react';
import { storiesOf } from '@storybook/react';

import { Input } from '../src';

export default () => {
  storiesOf('Input', module).add('Default', () => (
    <Input onChange={() => null} placeholder="Name" name="Name" />
  ));
};
