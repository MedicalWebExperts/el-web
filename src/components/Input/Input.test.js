import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';

const props = {
  name: 'input',
  type: 'text',
  placeholder: 'Input',
  onChange: () => null,
};

const customStyles = {
  marginTop: 30,
  lineHeight: 2,
};

describe('Component Input Snapshot', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => ({ matches: true })),
    });
  });

  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Input {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const tree = renderer.create(<Input {...props} styles={customStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
