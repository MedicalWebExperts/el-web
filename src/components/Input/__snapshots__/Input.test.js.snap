// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Input Snapshot added optional style inline prop 1`] = `
<div
  className="jsx-212918199"
>
  <input
    className="jsx-212918199"
    name="input"
    onChange={[Function]}
    placeholder="Input"
    style={
      Object {
        "lineHeight": 2,
        "marginTop": 30,
      }
    }
    type="text"
    value=""
  />
</div>
`;

exports[`Component Input Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-212918199"
>
  <input
    className="jsx-212918199"
    name="input"
    onChange={[Function]}
    placeholder="Input"
    style={Object {}}
    type="text"
    value=""
  />
</div>
`;
